packages/edit-post/src/components/meta-boxes/use-meta-box-initialization.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * WordPress dependencies
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { store as editPostStore } from '../../store';
+
+/**
+ * Initializes postboxes (wp core script) and meta box saving for all meta box locations.
+ *
+ * @param { boolean } enabled
+ */
+export default ( enabled ) => {
+	const { areInitialized, isEditorReady, hasAny } = useSelect(
+		( select ) => {
+			if ( ! enabled ) {
+				return {};
+			}
+			const { __unstableIsEditorReady } = select( editorStore );
+			const { areMetaBoxesInitialized, getMetaBoxesPerLocation } =
+				select( editPostStore );
+			return {
+				areInitialized: areMetaBoxesInitialized(),
+				isEditorReady: __unstableIsEditorReady(),
+				hasAny:
+					getMetaBoxesPerLocation( 'normal' ).length > 0 ||
+					getMetaBoxesPerLocation( 'advanced' ).length > 0 ||
+					getMetaBoxesPerLocation( 'side' ).length > 0,
```

Comment:
Could this length checks be avoided by using `isMetaBoxLocationActive` directly?

Reply:
Yes! Thanks for the suggestion. It spurred me to reexamine this and find that it can be simplified further. Since all locations are tested then calling the `hasMetaBoxes` selector would be most succinct. Yet, the `Layout` is already using that selector for a couple conditions and so can just pass it in to this hook.

Furthermore, it seems testing `areMetaBoxesInitialized` is extraneous, the `initializeMetaBoxes` action already returns early if initialized so it doesn’t help to test it twice.

Thus I’ve pushed db46c07d5e0f4a87db9784fb7143b6b976e0947e with those changes.